src/main/scala/uk/gov/hmrc/ct/computations/CP668.scala

```diff
@@ -16,9 +16,49 @@
 
 package uk.gov.hmrc.ct.computations
 
-import uk.gov.hmrc.ct.box.{CtBoxIdentifier, CtOptionalInteger, Input}
+import uk.gov.hmrc.ct.box.{CtValidation, CtBoxIdentifier, CtOptionalInteger, Input}
+import uk.gov.hmrc.ct.computations.Validators.ComputationValidatableBox
+import uk.gov.hmrc.ct.computations.calculations.MachineryAndPlantCalculator
+import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 
-case class CP668(value: Option[Int]) extends CtBoxIdentifier(name = "Writing down allowance claimed from special rate pool") with CtOptionalInteger with Input
+import scala.math.BigDecimal.RoundingMode
+
+case class CP668(value: Option[Int]) extends CtBoxIdentifier(name = "Writing down allowance claimed from special rate pool") with CtOptionalInteger with Input with ComputationValidatableBox[ComputationsBoxRetriever] with MachineryAndPlantCalculator {
+  override def validate(boxRetriever: ComputationsBoxRetriever) = {
+    validateZeroOrPositiveInteger(this) ++
+      fieldhasValueWhenTrading(boxRetriever, "CP668", value) ++
+      specialRatePoolClaimedNotGreaterThanMaxSpecialPool(boxRetriever)
+  }
+
+  private def specialRatePoolClaimedNotGreaterThanMaxSpecialPool(retriever: ComputationsBoxRetriever): Set[CtValidation] = {
+    val maxSP = Math.max(0, calcSpecialRateAllowance(retriever))
+
+    value match {
+          case Some(writingDownAllowanceClaimedFromSpecialRatePool) =>
```

Comment:
Just reformmated it, the case statement is fixes, the indentation of the ++ remains.